### PR TITLE
Adding script to build Docker images with ETA installed

### PR DESCRIPTION
**Pull request by brimoor**
This PR adds a script for building Docker images with ETA installed.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        wget \
+        python3.6 \
+        python3.6-dev \
+        libcupti-dev \
+        ffmpeg \
+        imagemagick \
+    && ln -s /usr/bin/python3.6 /usr/local/bin/python \
+    && curl https://bootstrap.pypa.io/get-pip.py | python
+
+WORKDIR /dev
```
Comment (ecrews) [inline docker/Dockerfile]: Here's your problem - `/dev` is a `tmpfs`: https://superuser.com/questions/675568/shrink-or-delete-udev-partition-in-ubuntu/675664#675664

Change the working directory something like `/usr/src/app` instead.

File: docker/build.bash [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+[ ${SHOW_HELP} = true ] && usage && exit 0
+
+# Parse positional arguments
+shift $((OPTIND - 1))
+BASE_IMAGE=$1
+TENSORFLOW_VERSION=$2
+TAG=$3
+
+# Clone ETA
+git clone https://github.com/voxel51/eta
```
Comment (ecrews) [inline docker/build.bash]: Not sure this is desirable - the usual pattern is that the Dockerfile is in the root of the project directory and you copy the code in.

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This folder provides support for building
+[Docker images](https://www.docker.com) with pre-installed ETA environments.
+
+
+## Dependencies
+
+- [Docker Community Edition](https://hub.docker.com/search/?type=edition&offering=community)
+- To build/run GPU-enabled images, you must install `nvidia-docker` by
+following [these instructions](https://github.com/NVIDIA/nvidia-docker)
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/README.md]: +1 issue made for this: https://github.com/voxel51/eta/issues/443

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+
+## Using an image
+
+After you have built an ETA image, you can run the image with an interactive
+shell as follows:
+
+```shell
+# For CPU images
+docker run -it --entrypoint=/bin/bash $TAG
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/README.md]: +1 issue made for this: https://github.com/voxel51/eta/issues/444

[the remark this answers is not in the document]
Comment (brimoor) [reply docker/README.md]: got it

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+# version to function properly, and some commands here seem to mess with the
+# numpy version installed via the `requirements.txt` file
+#
+
+RUN apt-get update \
+    && apt-get -y --no-install-recommends install software-properties-common \
+    && add-apt-repository -y ppa:deadsnakes/ppa \
+    && apt-get update \
+    && apt-get -y --no-install-recommends install \
+        sudo \
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/Dockerfile]: +1 issue made for this: https://github.com/voxel51/eta/issues/444

File: docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        ffmpeg \
+        imagemagick \
+    && ln -s /usr/bin/python3.6 /usr/local/bin/python \
+    && curl https://bootstrap.pypa.io/get-pip.py | python
+
+WORKDIR /dev
+COPY eta/ eta/
+ARG TENSORFLOW_VERSION
+RUN pip --no-cache-dir  install --upgrade pip setuptools \
+    && pip --no-cache-dir install -r eta/requirements.txt \
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/Dockerfile]: Agree, but most of the extra work is non-Python related install, so there's a bit of work required to make this as clean as possible

File: docker/build.bash [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+fi
+git submodule init
+git submodule update
+cp config-example.json config.json
+rm -rf .git/modules  # trim the fat (removes ~500MB)
+cd ..
+
+# Build image
+docker build \
+    --file Dockerfile \
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/build.bash]: removed

File: docker/build.bash [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+# Clone ETA
+git clone https://github.com/voxel51/eta
+cd eta
+if [[ ! -z $ETA_BRANCH_OR_COMMIT ]]; then
+    git checkout $ETA_BRANCH_OR_COMMIT
+fi
+git submodule init
+git submodule update
+cp config-example.json config.json
+rm -rf .git/modules  # trim the fat (removes ~500MB)
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/build.bash]: +1 done

File: docker/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+#### GPU example
+
+The following snippet builds a GPU-enabled image on Ubuntu 16.04 with CUDA 9.0:
+
+```shell
+BASE_IMAGE="nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04"
+TENSORFLOW_VERSION="tensorflow-gpu==1.12.0"
+TAG="eta-gpu-ubuntu-16.04-cuda-9.0"
```
[the remark this answers is not in the document]
Comment (brimoor) [reply docker/README.md]: +1 done

File: docker/Dockerfile (hunk 1 above)
Comment (brimoor) [reply docker/Dockerfile]: AHHHHHHH. Thanks for catching

File: docker/build.bash (hunk 2 above)
Comment (brimoor) [reply docker/build.bash]: thanks for the rec- updated

**Comment by brimoor** [comment]
@ecrews updated and now works. Take a look!

**Comment by ecrews** [comment]
Doing a second pass now!

**Comment by brimoor** [comment]
@ecrews thanks for your help! If there's anything else on your wishlist, it would be nice to track via Issues

**Pull request merged**